pkg/bucketaccessrequest/bucketaccessrequest.go

```diff
@@ -131,6 +143,10 @@ func (b *bucketAccessRequestListener) provisionBucketAccess(ctx context.Context,
 		return err
 	}
 
+	if !util.CheckFinalizer(bucketAccessRequest, util.BARDeleteFinalizer) {
```

Comment:
Why can't the finalizer be added to the `bucketAccess.Metadate` above, before the `Create` call? If this works, then you don't need to retry finalizer update block below.

Reply:
Finalizers to BA are addressed in the Provisioner as discussed.

Reply:
Ok, so are you saying the central controller does not add finalizers to the BA? What is confusing to me is why is the code checking for a finalizer here right after it creates a new BA?